.github/workflows/rust.yml

```diff
@@ -15,8 +15,34 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Build
-      run: cargo build --verbose
-    - name: Run tests
-      run: cargo test --verbose
+      - uses: actions/checkout@v3
+
+      - name: Install Rust nightly
+        run: |
+          curl -sSf https://rust-lang.org/install.sh | sh
+          rustup default nightly
+
+      - name: Install cargo-audit
+        run: cargo install cargo-audit
+
+      - name: Check for vulnerabilities
+        run: cargo audit
+
+      - name: Format code
+        run: |
+          rustup component add rustfmt
+          cargo fmt --all -- --check
+        
+      - name: Run Clippy linter
+        run: |
+          rustup component add clippy
+          cargo clippy -- -D warnings
```

Comment:
The issue here is that it is going to force some areas in the code to fail this warning, but functionally they are designed that way for a reason

Reply:
please give me some time alex i will work on this

Reply:
@AlexsJones i used this command `cargo clippy -- -D warnings` to deny all the lint errors that will stop the workflow hoping it would make the codebase clean . But i get your point, some things are designed that way for a specific reason right .

So instead of denying all the lint errors we can modify it in such a way that it will only denies specific lint using the command `cargo clippy -- -A clippy::all -D clippy::absurd_extreme_comparisons`. But i think whitelisting each lints by name is a time consuming task and it wont be effective . So for this, we can make use of the lint groups that are available in [clippy](https://github.com/rust-lang/rust-clippy#clippy) but even still this might be a roadblock for devs because it will throw errors .

Kindly share your thoughts on this whether we can remove this clippy feature or we can whilelist particular lints (by name or by group) and i will make the necessary changes alex.

PS : [for your reference list of lints available in clippy](https://rust-lang.github.io/rust-clippy/master/)